app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java

```diff
@@ -168,7 +166,9 @@ public void queryPairedDevices() {
 
         this.pairedDevices = this.bluetoothAdapter.getBondedDevices();
         for (final BluetoothDevice device : this.pairedDevices) {
-            this.pairedDevicesArrayAdapter.add(device.getName() + "\n" + device.getAddress());
+            if (this.pairedDevicesArrayAdapter.getPosition(device.getName() + "\n" + device.getAddress()) == -1) {
```

Comment:
I found this weird as well, but getPosition() returns -1 when it can't find the entry.